Sage/carbon
# chore: remove deprecated checkbox

### Proposed behaviour

- Classic theme has been removed from `configurable-items` because it uses the `__deprecated__/checkbox` internally.
- `__deprecated__/checkbox` has been removed

### Current behaviour
We have a deprecated and an experimental checkbox.
`configurable-items` supports classic and DLS themes.

### Checklist


<del>- [ ] Screenshots are included in the PR
<del>- [ ] Carbon implementation and Design System documentation are congruent
<del>- [ ] All themes are supported

<del>- [ ] Cypress automation tests added or updated


<del>- [ ] Typescript `d.ts` file added or updated

### Additional context
~Blocked by #2985~

### Testing instructions


- Ensure that the checkboxes in storybook still work. Dialog stories now use the `__experimental__` one so there will be a mix of themes in the Dialog story.

- Ensure that the classic theme has been removed from `configurable-items` and that the DLS themes still look the same.
